src/libcephfs.cc

```diff
@@ -2075,11 +2075,10 @@ extern "C" void ceph_finish_reclaim(class ceph_mount_info *cmount)
   cmount->get_client()->finish_reclaim();
 }
 
-extern "C" void ceph_ll_register_callbacks(class ceph_mount_info *cmount,
-					   struct ceph_client_callback_args *args)
+extern "C" int ceph_ll_register_callbacks(class ceph_mount_info *cmount,
+					  struct ceph_client_callback_args *args)
```

Comment:
I know I suggested this, but now I'm wondering if this is a sane thing to do. One potentially problematic scenario:

Suppose someone builds a binary that links to this lib (with the int return), and then at runtime loads an older version of the library that has this symbol but that returns void. The void return function likely won't load the return value register, and we'll end up with weird runtime errors there. It'd be better if the symbol were to fail to resolve, I think.

@batrick, @lxbsz : thoughts?

Reply:
Yeah, the more I think about it, if we wanted to do this, we'd need to bump the soname for the library. That's probably a bigger hassle than just adding a ceph_ll_register_callbacks2.

Reply:
This is rare case though. I have test it and didn't hit any issue till now.

Reply:
Checked the assembly code, for exmaple when calling the callback funcion the binary code will use the `%eax` as the return value register, but if the binary loads an old `libcephfs.so`, the `%eax` register in the callback function will be a random value in the end, which will be used for other purpose, not as the return value register.

So this will certainly be a problem in this case. Let me add the callback2 back.